typespec-extension/src/code-model-builder.ts

```diff
@@ -463,11 +460,17 @@ export class CodeModelBuilder {
         }
       }
 
-      // Internal on Anonymous
-      if (schemaUsage?.includes(SchemaContext.Anonymous)) {
-        const index = schemaUsage.indexOf(SchemaContext.Internal);
-        if (index < 0) {
-          schemaUsage.push(SchemaContext.Internal);
+      // Internal on PublicSpread, but Public takes precedence
+      if (schemaUsage?.includes(SchemaContext.PublicSpread)) {
+        // remove PublicSpread as it served its purpose
+        schemaUsage.splice(schemaUsage.indexOf(SchemaContext.PublicSpread), 1);
+
+        // Public would override PublicSpread, hence do nothing if this schema is Public
+        if (!schemaUsage?.includes(SchemaContext.Public)) {
+          const index = schemaUsage.indexOf(SchemaContext.Internal);
+          if (index < 0) {
```

Comment:
I'm a bit confused, What does `index < 0` mean? `schemaUsage` include `SchemaContext.Internal` or not include `SchemaContext.Internal` ? Can we use `schemaUsage?.includes(SchemaContext.Internal)` instead as it is more straightforward?

Reply:
Yes, it is for `includes`. Changed.